src/app/components/algorithms/algorithm-view/algorithm-view.component.ts

```diff
@@ -51,37 +55,62 @@ export class AlgorithmViewComponent implements OnInit, OnDestroy {
     private route: ActivatedRoute,
     private utilService: UtilService,
     private config: ApiConfiguration,
-    public guard: ChangePageGuard
+    public guard: ChangePageGuard,
+    private planqkPlatformLoginService: PlanqkPlatformLoginService
   ) {}
 
   ngOnInit(): void {
-    this.routeSub = this.route.params.subscribe(({ algoId }) => {
-      this.algorithmService.getAlgorithm({ algorithmId: algoId }).subscribe(
-        (algo: AlgorithmDto) => {
-          this.algorithm = algo;
-          this.frontendAlgorithm = JSON.parse(
-            JSON.stringify(algo)
-          ) as AlgorithmDto;
-          let subheading = this.algorithm.computationModel
-            .toString()
-            .toLowerCase();
-          subheading = subheading[0].toUpperCase() + subheading.slice(1);
-          this.links[0] = {
-            heading: this.createBreadcrumbHeader(this.algorithm),
-            subHeading: subheading + ' Algorithm',
-          };
-          this.getApplicationAreasForAlgorithm(algoId);
-          this.getProblemTypesForAlgorithm(algoId);
-          this.getTagsForAlgorithm(algoId);
-          this.fetchRevisions();
-        },
-        () => {
+    this.dataSubscription = this.planqkPlatformLoginService
+      .isLoggedIn()
+      .pipe(
+        mergeMap((loggedIn) => {
+          if (loggedIn) {
+            // Try planqk
+            this.configuredApiUrl = this.config.rootUrl;
+            this.config.rootUrl = 'https://platform.planqk.de/qc-catalog';
+          }
+          return this.loadAlgorithm();
+        }),
+        catchError(() => {
+          // Not available on PlanQK: Try on qc-atlas api
+          this.config.rootUrl = this.configuredApiUrl;
+          return this.loadAlgorithm();
+        }),
+        catchError(() => {
           this.utilService.callSnackBar(
             'Error! Algorithm could not be retrieved.'
           );
-        }
-      );
-    });
+          return of();
+        })
+      )
+      .subscribe();
+  }
+
+  loadAlgorithm(): Observable<void> {
+    return this.route.params.pipe(
+      mergeMap(({ algoId }) =>
+        this.algorithmService.getAlgorithm({ algorithmId: algoId })
+      ),
+      map((algo: AlgorithmDto) => {
```

Comment:
To avoid side effects in the map, you could put the contents of the map in the `subscribe` in line 86. (Or in a separate class function that is called in the `subscribe`, but be careful with references to `this` in this case, for normal functions, this does not behave like expected, but for lambda functions it does!)
For this, the second error handler (line 79) would also need to be moved into the subscribe function.